lib/galaxy/tools/actions/__init__.py

```diff
@@ -96,14 +96,15 @@ def process_dataset(data, formats=None):
                 # Checked security of whole collection all at once if mapping over this input, else
                 # fetch dataset details for this input from the database.
                 if collection_info and collection_info.is_mapped_over(input_name):
-                    action_tuples = collection_info.map_over_action_tuples(input_name)
-                    if not trans.app.security_agent.can_access_datasets(current_user_roles, action_tuples):
-                        raise ItemAccessibilityException("User does not have permission to use a dataset provided for input.")
+                    if not trans.user_is_admin:
+                        action_tuples = collection_info.map_over_action_tuples(input_name)
+                        if not trans.app.security_agent.can_access_datasets(current_user_roles, action_tuples):
+                            raise ItemAccessibilityException("User does not have permission to use a dataset provided for input.")
                     for action, role_id in action_tuples:
```

Comment:
It seems to me that `action_tuples` is not defined when the user is an admin?

Reply:
You're absolutely right. Thanks for the catch!